src/Runtime/Eval.hs

```diff
@@ -91,36 +91,23 @@ evalBinOp Div l r      = evalNumOp "/" div l r
 evalBinOp Mod l r      = evalNumOp "%" mod l r
 evalBinOp Less l r     = evalCompareOpInt (<) l r
 evalBinOp Leq l r      = evalCompareOpInt (<=) l r
-evalBinOp NotEquiv l r = evalNotEquiv l r
-evalBinOp Equiv l r    = evalEquiv l r
+evalBinOp Equiv l r    = evalEq (==) l r
+evalBinOp NotEquiv l r = evalEq (/=) l r
 evalBinOp Geq l r      = evalCompareOpInt (>=) l r
 evalBinOp Greater l r  = evalCompareOpInt (>) l r
 evalBinOp Get l r      = do
    board <- eval l
    pos   <- eval r
    case (board, pos) of
       (Vboard arr, Vt [Vi x, Vi y]) -> return $ tryUnsafeBoardAccess (x,y) arr
-      _ -> return $ Err $ "Could not access" ++ show l ++ " in " ++ show "r"
-
--- | evaluates the /= operation
-evalNotEquiv l r = do
-   eq <- evalEquiv l r
-   case eq of
-      (Vb b) -> return $ Vb (not b)
-      eq     -> return eq
-
--- | evaluates the == operation
-evalEquiv :: (Expr a) -> (Expr a) -> Eval Val
-evalEquiv l r = do
-   v1 <- eval l
-   v2 <- eval r
-   case (v1,v2) of
-     (Vi l', Vi r') -> return $ Vb (l' == r')
-     (Vs l', Vs r') -> return $ Vb (l' == r')
-     (Vb l', Vb r')   -> return $ Vb (l' == r')
-     (Vt l', Vt r')   -> return $ Vb (l' == r')
-     (Vboard l', Vboard r') -> return $ Vb (l' == r')
-     _              -> return $ Err $ "Could not compare " ++ (show l) ++ " to " ++ (show r)
```

Comment:
The type checker catches these other cases, with the exception of symbol comparisons. One case we probably don't want to allow is board-symbol comparison, but I think that the appropriate place to change that is in the type unification, so it will be in a separate change set.